dandiapi/api/views/version.py

```diff
@@ -131,6 +135,18 @@ def update(self, request, **kwargs):
                     metadata=locked_version.metadata,
                 )
 
+                # For unpublished dandisets, update or create the draft DOI
+                # to keep it in sync with the latest metadata
+                if not locked_version.dandiset.embargoed:
+                    transaction.on_commit(
+                        lambda: update_draft_version_doi_task.delay(locked_version.id)
+                    )
+                else:
+                    logger.debug(
+                        'Skipping DOI update for embargoed Dandiset %s.',
+                        locked_version.dandiset.identifier,
+                    )
```

Comment:
If we're going to be keeping doi metadata up to date anytime a draft version is updated, then I'm even more in favor of what I proposed in my other comment (a table for keeping track of this). Doing it in this fashion is just a disaster waiting to happen in regards to synchronization of metadata.